needle/modules/configuration/mdm_assess.py

```diff
@@ -0,0 +1,153 @@
+from core.framework.module import BaseModule
+from core.framework.framework import FrameworkException
+from core.device.device import Device
+from core.utils.utils import Utils
+from core.utils.constants import Constants
+from plistlib import readPlist,_InternalDict
+
+class Module(BaseModule):
+    meta = {
+        'name': 'MDM Assess Effective',
+        'author': 'Oliver Simonnet (@MWRLabs)',
+        'description': 'Configuration Assessment Module',
+        'options': (
+            ('template', True, True, 'Configuration template. [Plist]'),
+            ('output', True, True,  'Full path of the output folder.'),
+            ('pull_only', False, False, 'Only save the configuration file')
+        ),
+        'comments': [
+            'TEMPLATE: This is an EffectiveUserSettings.plist '\
+            'file specifying the desired configuration.',
+            'PULL_ONLY: Pulls the configuration from the device and '\
+            'stores it locally. Used if you only wish to obtain a '\
+            'copy of the configuration file and not perform an assessmen']
+    }
+
+    # ==================================================================================================================
+    # UTILS
+    # ==================================================================================================================
+    def __init__(self, params):
+        BaseModule.__init__(self, params)
+        # Setting default output file
+        self.options['output'] = self._global_options['output_folder']
+
+
+    # Pulls a file from the device and returns the local file name
+    def save_file(self, remote_name, local_name):
+        if not self.options['output']:
+            return
+        temp_name = 'MDM_Assess_Device{}'.format(local_name)
+        local_name = self.local_op.build_output_path_for_file(temp_name, self)
+        self.device.pull(remote_name, local_name)
+        return local_name
+
+
+    # Parse Plist configuration data into dict
+    def structure_data(self, config_file):
+        try:
+            readPlist(config_file)
+            config, merged = readPlist(config_file), {}
+            for k in config.keys(): 
+                merged.update(config[k])
+            return merged
+        except:
+            # If not, print error and exit
+            self.printer.error('Invalid configuration file!')
+            self.printer.verbose('Invalid file: %s' % config_file)
+            raise FrameworkException()
```

Comment:
when you raise a `FrameworkException`, you should provide an error message.
this could be:

```
self.printer.error('Invalid file: %s' % config_file)
raise FrameworkException('Invalid configuration file!')
```